server/http_instrumentation.go

```diff
@@ -85,15 +83,18 @@ func (m instrumentedHandlerFactory) InitializeMetrics(labels prometheus.Labels)
 
 // NewHandler creates a new instrumented HTTP handler with the given extra labels and calling the "next" handlers.
 func (m instrumentedHandlerFactory) NewHandler(extraLabels prometheus.Labels, next http.Handler) http.HandlerFunc {
-	return func(w http.ResponseWriter, r *http.Request) {
-		m.labelsMutex.Lock()
-		defer m.labelsMutex.Unlock()
+	// Default group and handler to "unknown" if no extra labels are provided as a parameter.
+	if extraLabels == nil {
+		extraLabels = prometheus.Labels{"group": "unknown", "handler": "unknown"}
+	}
 
-		// Default group and handler to "unknown" if no extra labels are provided as a parameter.
-		if extraLabels == nil {
-			extraLabels = prometheus.Labels{"group": "unknown", "handler": "unknown"}
+	return func(w http.ResponseWriter, r *http.Request) {
+		extraLabelsCopy := make(prometheus.Labels, len(extraLabels))
```

Comment:
What about a more "speaking name"?

```suggestion
		requestLabels := make(prometheus.Labels, len(extraLabels))
```